cmd/assetsvc/postgresql_utils.go

```diff
@@ -71,12 +72,30 @@ func (m *postgresAssetManager) getPaginatedChartList(namespace, repo string, pag
 		repoQuery = strings.Join(clauses, " AND ")
 		repoQuery = "WHERE " + repoQuery
 	}
-	dbQuery := fmt.Sprintf("SELECT info FROM %s %s ORDER BY info ->> 'name' ASC", dbutils.ChartTable, repoQuery)
+
+	paginationClause := ""
+	if pageSize > 0 && pageNumber > 0 {
+		offset := (pageNumber - 1) * pageSize
+		paginationClause = fmt.Sprintf("LIMIT %d OFFSET %d", pageSize, offset)
+	}
+
+	dbQuery := fmt.Sprintf("SELECT info FROM %s %s ORDER BY info ->> 'name' ASC %s", dbutils.ChartTable, repoQuery, paginationClause)
 	charts, err := m.QueryAllCharts(dbQuery, queryParams...)
 	if err != nil {
-		return nil, 0, err
+		return nil, 0, 0, err
+	}
+
+	dbCountQuery := fmt.Sprintf("SELECT count(info) FROM %s %s", dbutils.ChartTable, repoQuery)
+	count, err := m.QueryCount(dbCountQuery, queryParams...)
+	if err != nil {
+		return nil, 0, 0, err
 	}
```

Comment:
this is calculating the total number of charts, right? (regardless of the pagination). If that's the case, you just need to do the query in case the pagination is requested (pageSize > 0) so I would do this query within the `if` below (removing the `count > 0` condition).

Reply:
Mmm, I see your point. We can skip this query as you mention, with no pagination this totalCharts would be just `len(charts)`. I'll address that.